scvi/inference/posterior.py

```diff
@@ -43,13 +36,39 @@
 )
 from scvi.models.distributions import NegativeBinomial, ZeroInflatedNegativeBinomial
 from scipy.stats import spearmanr
+from scvi.dataset._constants import (
+    _X_KEY,
+    _BATCH_KEY,
+    _LOCAL_L_MEAN_KEY,
+    _LOCAL_L_VAR_KEY,
+    _LABELS_KEY,
+)
 
 logger = logging.getLogger(__name__)
 
 
-class SequentialSubsetSampler(SubsetRandomSampler):
+class BatchSampler(torch.utils.data.sampler.Sampler):
```

Comment:
add docstring and explain why we ended up doing this versus standard PyTorch Streamline of things